### PR TITLE
feat: enable azure pg connectivity

### PR DESCRIPTION
**Pull request by vobu**
enhancing the `credentials` obj by a boolean value to be able to properly connect to a pg db running on azure.

also added tests for major connectivity scenarios (docker, btp, azure).

### CODE REVIEW

**Review by patricebender — COMMENTED** [review]
Thanks Volker, could you also add a changelog entry? Thanks!

**Comment by vobu** [comment]
> Thanks Volker, could you also add a changelog entry? Thanks!

done - not sure if i did it right though...put placeholders for `semver` and the release date 🤔

File: postgres/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@
 - The format is based on [Keep a Changelog](http://keepachangelog.com/).
 - This project adheres to [Semantic Versioning](http://semver.org/).
 
+## Version x.x.x - yyyy-MM-dd
+
+### Added
+
+- connectivity to Azure PostGreSQL
```
Comment (patricebender) [inline postgres/CHANGELOG.md:11]:
```suggestion
- connectivity to Azure PostgreSQL
```

**Comment by patricebender** [comment]
> > Thanks Volker, could you also add a changelog entry? Thanks!
> 
> done - not sure if i did it right though...put placeholders for `semver` and the release date 🤔

looks good to me. In the future, we could also open separate branches for the changelogs, i.e. `changelog/sqlite|postgres|db-service` - that way we won't get merge conflicts

**Comment by vobu** [comment]
we could also start enforcing commit msg formats. then generate a changelog from them. this is how we do it with [wdi5](https://github.com/ui5-community/wdi5/blob/main/CHANGELOG.md) and it's working well. 
I'll open a discussion and see what the others think about it.

**Review by BobdenOs — APPROVED** [review]

**Pull request merged**